### ISSUE TITLE
EDSC-994: Add PSA CORRECTIONACHIEVED to Earthdata Search filter for AST_L1T

### ISSUE THREAD

**Issue by earthdata-tasktop** [nasa/edsc-search-test-config-3#198]
Add PSA CORRECTIONACHIEVED to Earthdata Search filter for AST_L1T

The AST_L1T product can have one of four levels of correction. To assist users who may only be interested in scenes with a specific level of correction, we are requesting the addition of a search filter for the level of CORRECTIONACHIEVED within the Earthdata Search client.

**Comment by earthdata-tasktop** [comment]
(Comment from Matt Martens):

The CORRECTION ACHIEVED PSA has been added to EDSC as a filter option.  This issue can be closed.

**Comment by earthdata-tasktop** [comment]
(Comment from Francell Gray):

Verified in UAT that the additional attributes for the AST_L1T collection are available on the Granule Filter page (see screen shots).

Note: Had to use the URL search.uat.earthdata.nasa.gov/search/granules?labs=true because there apparently still has not a final determination on the additional attribute implementation.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Steps to Verify

* Use AST_L1T collection and go to granule list page.
* Open granule filter panel
* You should see additional attributes listed, ***if the collection is properly tagged in CMR***

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

I've attached a file showing part of the situation. These attributes are just a wall of fields (note the scrolbar position). AST_L1T has 46 of them.

The one they want us to enable will say this:

> CorrectionAchieved  

> The identifier to inform the user of the level of correction achieved for the product. (The choices are: Terrain+Precision, Terrain+Systematic, Systematic, Precision)  

> [String Value                                            ]

Users will need to manually type in one of the 4 listed values to get the desired filter.

There are well over a dozen fields that amount to booleans and ought to be checkboxes, but instead instruct the user to type in the string literals "Yes, band is acquired" or "No, band was not acquired".

We could absolutely make this work with a series of one-off hacks, but I'd very much prefer to go after more sustainable solutions, such as UMM updates, metadata fixes, and CMR enhancements to determine valid values for AAs.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Provide for just AST_L1T

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

We could use some input here. The way we implemented this, we added the ability to selectively enable additional attributes on a per-collection level. It turns out AST*L1T has roughly the same additional attributes as AST*L1A, which is an enormous list with numerous usability problems. See the following link, click on the pencil icon, and scroll down: [https://search.earthdata.nasa.gov/search/granules?labs=true&amp;amp;p=C14758250-LPDAAC_ECS](https://search.earthdata.nasa.gov/search/granules?labs=true&p=C14758250-LPDAAC_ECS)

There is a significant amount of unusable fields we'd be forced to present. Try to figure out what to put into the Band1*Available field, for instance (I've yet to figure it out). AST*L1A was the primary reason we hid these attributes in the first place.

At this point, we could in theory update our code to pick out the one specific additional attribute requested. Out of principle, though, I really don't like this sort of one-off code that only applies to a single collection, especially when we have an avenue to accomplish what the provider wants through providing better metadata.

So, do we hack it? Do we enable all AA's and swallow the usability problems? Do we kick this back to the provider?

**Comment by earthdata-tasktop** [comment]
(Comment from Francell Gray):

Verified in UAT that the additional attributes for the AST_L1T collection are available on the Granule Filter page (see screen shots).

Note: Had to use the URL search.uat.earthdata.nasa.gov/search/granules?labs=true because there apparently still has not a final determination on the additional attribute implementation.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Steps to Verify

* Use AST_L1T collection and go to granule list page.
* Open granule filter panel
* You should see additional attributes listed, ***if the collection is properly tagged in CMR***

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

I've attached a file showing part of the situation. These attributes are just a wall of fields (note the scrolbar position). AST_L1T has 46 of them.

The one they want us to enable will say this:

> CorrectionAchieved  

> The identifier to inform the user of the level of correction achieved for the product. (The choices are: Terrain+Precision, Terrain+Systematic, Systematic, Precision)  

> [String Value                                            ]

Users will need to manually type in one of the 4 listed values to get the desired filter.

There are well over a dozen fields that amount to booleans and ought to be checkboxes, but instead instruct the user to type in the string literals "Yes, band is acquired" or "No, band was not acquired".

We could absolutely make this work with a series of one-off hacks, but I'd very much prefer to go after more sustainable solutions, such as UMM updates, metadata fixes, and CMR enhancements to determine valid values for AAs.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Provide for just AST_L1T

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

We could use some input here. The way we implemented this, we added the ability to selectively enable additional attributes on a per-collection level. It turns out AST*L1T has roughly the same additional attributes as AST*L1A, which is an enormous list with numerous usability problems. See the following link, click on the pencil icon, and scroll down: [https://search.earthdata.nasa.gov/search/granules?labs=true&amp;amp;p=C14758250-LPDAAC_ECS](https://search.earthdata.nasa.gov/search/granules?labs=true&p=C14758250-LPDAAC_ECS)

There is a significant amount of unusable fields we'd be forced to present. Try to figure out what to put into the Band1*Available field, for instance (I've yet to figure it out). AST*L1A was the primary reason we hid these attributes in the first place.

At this point, we could in theory update our code to pick out the one specific additional attribute requested. Out of principle, though, I really don't like this sort of one-off code that only applies to a single collection, especially when we have an avenue to accomplish what the provider wants through providing better metadata.

So, do we hack it? Do we enable all AA's and swallow the usability problems? Do we kick this back to the provider?

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Steps to Verify

* Use AST_L1T collection and go to granule list page.
* Open granule filter panel
* You should see additional attributes listed, ***if the collection is properly tagged in CMR***

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

I've attached a file showing part of the situation. These attributes are just a wall of fields (note the scrolbar position). AST_L1T has 46 of them.

The one they want us to enable will say this:

> CorrectionAchieved  

> The identifier to inform the user of the level of correction achieved for the product. (The choices are: Terrain+Precision, Terrain+Systematic, Systematic, Precision)  

> [String Value                                            ]

Users will need to manually type in one of the 4 listed values to get the desired filter.

There are well over a dozen fields that amount to booleans and ought to be checkboxes, but instead instruct the user to type in the string literals "Yes, band is acquired" or "No, band was not acquired".

We could absolutely make this work with a series of one-off hacks, but I'd very much prefer to go after more sustainable solutions, such as UMM updates, metadata fixes, and CMR enhancements to determine valid values for AAs.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Provide for just AST_L1T

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Provide for just AST_L1T

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

We could use some input here. The way we implemented this, we added the ability to selectively enable additional attributes on a per-collection level. It turns out AST*L1T has roughly the same additional attributes as AST*L1A, which is an enormous list with numerous usability problems. See the following link, click on the pencil icon, and scroll down: [https://search.earthdata.nasa.gov/search/granules?labs=true&amp;amp;p=C14758250-LPDAAC_ECS](https://search.earthdata.nasa.gov/search/granules?labs=true&p=C14758250-LPDAAC_ECS)

There is a significant amount of unusable fields we'd be forced to present. Try to figure out what to put into the Band1*Available field, for instance (I've yet to figure it out). AST*L1A was the primary reason we hid these attributes in the first place.

At this point, we could in theory update our code to pick out the one specific additional attribute requested. Out of principle, though, I really don't like this sort of one-off code that only applies to a single collection, especially when we have an avenue to accomplish what the provider wants through providing better metadata.

So, do we hack it? Do we enable all AA's and swallow the usability problems? Do we kick this back to the provider?

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

We could use some input here. The way we implemented this, we added the ability to selectively enable additional attributes on a per-collection level. It turns out AST*L1T has roughly the same additional attributes as AST*L1A, which is an enormous list with numerous usability problems. See the following link, click on the pencil icon, and scroll down: [https://search.earthdata.nasa.gov/search/granules?labs=true&amp;amp;p=C14758250-LPDAAC_ECS](https://search.earthdata.nasa.gov/search/granules?labs=true&p=C14758250-LPDAAC_ECS)

There is a significant amount of unusable fields we'd be forced to present. Try to figure out what to put into the Band1*Available field, for instance (I've yet to figure it out). AST*L1A was the primary reason we hid these attributes in the first place.

At this point, we could in theory update our code to pick out the one specific additional attribute requested. Out of principle, though, I really don't like this sort of one-off code that only applies to a single collection, especially when we have an avenue to accomplish what the provider wants through providing better metadata.

So, do we hack it? Do we enable all AA's and swallow the usability problems? Do we kick this back to the provider?

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

We could use some input here. The way we implemented this, we added the ability to selectively enable additional attributes on a per-collection level. It turns out AST*L1T has roughly the same additional attributes as AST*L1A, which is an enormous list with numerous usability problems. See the following link, click on the pencil icon, and scroll down: [https://search.earthdata.nasa.gov/search/granules?labs=true&amp;amp;p=C14758250-LPDAAC_ECS](https://search.earthdata.nasa.gov/search/granules?labs=true&p=C14758250-LPDAAC_ECS)

There is a significant amount of unusable fields we'd be forced to present. Try to figure out what to put into the Band1*Available field, for instance (I've yet to figure it out). AST*L1A was the primary reason we hid these attributes in the first place.

At this point, we could in theory update our code to pick out the one specific additional attribute requested. Out of principle, though, I really don't like this sort of one-off code that only applies to a single collection, especially when we have an avenue to accomplish what the provider wants through providing better metadata.

So, do we hack it? Do we enable all AA's and swallow the usability problems? Do we kick this back to the provider?

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

I've attached a file showing part of the situation. These attributes are just a wall of fields (note the scrolbar position). AST_L1T has 46 of them.

The one they want us to enable will say this:

> CorrectionAchieved  

> The identifier to inform the user of the level of correction achieved for the product. (The choices are: Terrain+Precision, Terrain+Systematic, Systematic, Precision)  

> [String Value                                            ]

Users will need to manually type in one of the 4 listed values to get the desired filter.

There are well over a dozen fields that amount to booleans and ought to be checkboxes, but instead instruct the user to type in the string literals "Yes, band is acquired" or "No, band was not acquired".

We could absolutely make this work with a series of one-off hacks, but I'd very much prefer to go after more sustainable solutions, such as UMM updates, metadata fixes, and CMR enhancements to determine valid values for AAs.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Provide for just AST_L1T

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Provide for just AST_L1T

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

We could use some input here. The way we implemented this, we added the ability to selectively enable additional attributes on a per-collection level. It turns out AST*L1T has roughly the same additional attributes as AST*L1A, which is an enormous list with numerous usability problems. See the following link, click on the pencil icon, and scroll down: [https://search.earthdata.nasa.gov/search/granules?labs=true&amp;amp;p=C14758250-LPDAAC_ECS](https://search.earthdata.nasa.gov/search/granules?labs=true&p=C14758250-LPDAAC_ECS)

There is a significant amount of unusable fields we'd be forced to present. Try to figure out what to put into the Band1*Available field, for instance (I've yet to figure it out). AST*L1A was the primary reason we hid these attributes in the first place.

At this point, we could in theory update our code to pick out the one specific additional attribute requested. Out of principle, though, I really don't like this sort of one-off code that only applies to a single collection, especially when we have an avenue to accomplish what the provider wants through providing better metadata.

So, do we hack it? Do we enable all AA's and swallow the usability problems? Do we kick this back to the provider?

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

We could use some input here. The way we implemented this, we added the ability to selectively enable additional attributes on a per-collection level. It turns out AST*L1T has roughly the same additional attributes as AST*L1A, which is an enormous list with numerous usability problems. See the following link, click on the pencil icon, and scroll down: [https://search.earthdata.nasa.gov/search/granules?labs=true&amp;amp;p=C14758250-LPDAAC_ECS](https://search.earthdata.nasa.gov/search/granules?labs=true&p=C14758250-LPDAAC_ECS)

There is a significant amount of unusable fields we'd be forced to present. Try to figure out what to put into the Band1*Available field, for instance (I've yet to figure it out). AST*L1A was the primary reason we hid these attributes in the first place.

At this point, we could in theory update our code to pick out the one specific additional attribute requested. Out of principle, though, I really don't like this sort of one-off code that only applies to a single collection, especially when we have an avenue to accomplish what the provider wants through providing better metadata.

So, do we hack it? Do we enable all AA's and swallow the usability problems? Do we kick this back to the provider?
